index.src.html

```diff
@@ -3384,53 +3408,92 @@
-    5. Assign |format| to {{VideoFrame/[[format]]}}
-    6. Assign |width| to the following attributes of |frame|:
-        {{VideoFrame/codedWidth}}, {{VideoFrame/displayWidth}}.
-    7. Assign |height| to the following attributes of |frame|:
-        {{VideoFrame/codedHeight}}, {{VideoFrame/displayHeight}}.
-    8. Assign «[ "x:" → `0`, "y" → `0`, "width" → |width|, "height" →
-        |height| ]» to |frame|.{{VideoFrame/visibleRect}}.
-    9. Assign `init`.{{VideoFrameInit/duration}} to
+    7. Assign |format| to {{VideoFrame/[[format]]}}
+    8. Assign |width| and |height| to |frame|'s {{VideoFrame/[[coded width]]}}
+        and {{VideoFrame/[[coded height]]}} respectively.
+    9. Let |defaultVisibleRect| be a new {{DOMRect}} constructed with
+        «[ "x:" → `0`, "y" → `0`, "width" → |width|, "height" → |height| ]»
+    10. Run the [=VideoFrame/Initialize Visible Rect and Display Size=]
+        algorithm with |init|, |frame|, |defaultVisibleRect|, |width|, and
+        |height|.
+    11. Assign `init`.{{VideoFrameInit/duration}} to
         |frame|.{{VideoFrame/duration}}.
-    10. Assign `init`.{{VideoFrameInit/timestamp}} to
+    12. Assign `init`.{{VideoFrameInit/timestamp}} to
         |frame|.{{VideoFrame/timestamp}}.
-    9. If |resource| has a known {{VideoColorSpace}}, assign its value to
+    13. If |resource| has a known {{VideoColorSpace}}, assign its value to
         {{VideoFrame/[[color space]]}}.
-    10. Otherwise, assign a new {{VideoColorSpace}}, constructed with an empty
+    14. Otherwise, assign a new {{VideoColorSpace}}, constructed with an empty
         {{VideoColorSpaceInit}}, to {{VideoFrame/[[color space]]}}.
 
+: <dfn for=VideoFrame>Initialize Visible Rect and Display Size</dfn> (with
+    |init|, |frame|, |defaultVisibleRect|, |defaultDisplayWidth| and
+    |defaultDisplayHeight|)
+:: 1. Let |visibleRect| be |defaultVisibleRect|.
+    2. If |init|.{{VideoFrameInit/visibleRect}} [=map/exists=], assign it to
+        |visibleRect|.
+    3. Assign |visibleRect|'s {{DOMRect/x}}, {{DOMRect/y}}, {{DOMRect/width}},
+        and {{DOMRect/height}}, to |frame|'s {{VideoFrame/[[visible left]]}},
+        {{VideoFrame/[[visible top]]}}, {{VideoFrame/[[visible width]]}}, and
+        {{VideoFrame/[[visible height]]}} respectively.
+    4. If {{VideoFrameInit/displayWidth}} and {{VideoFrameInit/displayHeight}}
+        [=map/exist=] in |init|, assign them to {{VideoFrame/[[display width]]}}
+        and {{VideoFrame/[[display height]]}} respectively.
+    5. Otherwise:
+        1. Let |widthScale| be the result of dividing |defaultDisplayWidth| by
+            |defaultVisibleRect|.{{DOMRect/width}}.
+        2. Let |heightScale| be the result of dividing |defaultDisplayHeight| by
+            |defaultVisibleRect|.{{DOMRect/height}}.
+        3. Multiply |frame|'s {{VideoFrame/[[visible width]]}} by |widthScale|
+            and round the result (away from zero). Assign the rounded result to
```

Comment:
I'm not sure about "(away from zero)", could be confusing. Can we avoid specifying?

Reply:
It's always positive, this can just say "up", but "away from zero" is quite a common phrasing (e.g. `man 3 round`).

Reply:
I don't think that interpretation is what was intended here though, I was expecting round-half-up.

Reply:
I've added "half" to phrase and added dfn for "half away from zero" that aims to clarify.

Reply:
I've added "half" to phrase and added dfn for "half away from zero" that aims to clarify.